Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -164,39 +166,41 @@ class OpenSearchParser {
             return nil
         }
 
-        var uiImage: UIImage?
-        var imageIndexers = docIndexer["Image"].all
+        let imageIndexers = docIndexer["Image"].all
+        var largestImage = 0
+        var largestImageElement: XMLElement?
 
+        // TODO: For now, just use the largest icon.
         for imageIndexer in imageIndexers {
-            // TODO: For now, we only look for 16x16 search icons.
-            let imageWidth = imageIndexer.element?.attributes["width"]
-            let imageHeight = imageIndexer.element?.attributes["height"]
-            if imageWidth?.toInt() != 16 || imageHeight?.toInt() != 16 {
-                continue
-            }
+            let imageWidth = imageIndexer.element?.attributes["width"]?.toInt()
+            let imageHeight = imageIndexer.element?.attributes["height"]?.toInt()
 
-            if imageIndexer.element?.text == nil {
+            // Only accept square images.
+            if imageWidth != imageHeight {
                 continue
             }
 
-            let imageURL = NSURL(string: imageIndexer.element!.text!)
-            if let imageURL = imageURL {
-                let imageData = NSData(contentsOfURL: imageURL)
-                if let imageData = imageData {
-                    let image = UIImage(data: imageData)
-                    if let image = image {
-                        uiImage = image
-                    } else {
-                        println("Error: Invalid search image data")
+            if let imageWidth = imageWidth {
+                if imageWidth > largestImage {
+                    if imageIndexer.element?.text != nil {
+                        largestImage = imageWidth
+                        largestImageElement = imageIndexer.element
                     }
-                } else {
-                    println("Error: Invalid search image data")
                 }
-            } else {
-                println("Error: Invalid search image data")
             }
         }
 
+        var uiImage: UIImage?
+
+        if let imageElement = largestImageElement,
+               imageURL = NSURL(string: imageElement.text!),
```

Comment:
This assumes imageLement.text is always present. Is that true? I guess malformed search engine xml files are not common. But it is easy to include it as another check in the `if let`.